app/controllers/SupplementaryUnitController.scala

```diff
@@ -53,15 +54,31 @@ class SupplementaryUnitController @Inject() (
         case Some(value) => form.fill(value)
       }
 
-      Ok(view(preparedForm, mode, recordId))
+      request.userAnswers.get(RecordCategorisationsQuery) match {
+        case None        => Redirect(routes.JourneyRecoveryController.onPageLoad().url)
+        case Some(query) =>
+          query.records(recordId).measurementUnit match {
+            case None                  => Ok(view(preparedForm, mode, recordId, ""))
```

Comment:
Do we want to load the page if there is no measurementUnit?

Reply:
Yes. measurementUnit can be null https://api.trade-tariff.service.gov.uk/green-lanes.html#get-green-lanes-goods-nomenclatures-id